Add a page on creating movies with Godot using Movie Maker functionality (#5889)
godotengine/godot-docs: tutorials/animation/creating_movies.rst

Output formats are provided by the MovieWriter class.
Godot has 2 built-in MovieWriters, and more can be implemented by extensions:

AVI (recommended)
^^^^^^^^^^^^^^^^^

AVI container with MJPEG for video and uncompressed audio. Features lossy video
compression, resulting in medium file sizes and fast encoding. The lossy
compression quality can be adjusted by changing
**Editor > Movie Writer > MJPEG Quality**.

The resulting file can be viewed in most video players, but it must be converted
to another format for viewing on the web or by Godot with the VideoStreamPlayer
node. MJPEG does not support transparency. AVI output is currently limited to a
file of 4 GB in size at most.

To use AVI, specify a path to an ``.avi`` file to be created in the
**Editor > Movie Writer > Movie File** project setting.

PNG
^^^

PNG image sequence for video and WAV for audio. Features lossless video
compression, at the cost of large file sizes and slow encoding. This is designed to be
:ref:`encoded to a video file with an external tool after recording <doc_creating_movies_converting_avi>`.
Transparency is currently not supported, even if the root viewport is set to be transparent.

> To use PNG, specify a ``.png` file to be created in the

Reviewer: ```suggestion
To use PNG, specify a ``.png`` file to be created in the
```

        if OS.has_feature("movie"):
            # When recording a movie, improve SDFGI cell density
            # without decreasing its maximum distance.
            get_viewport().world_3d.environment.sdfgi_min_cell_size *= 0.25
            get_viewport().world_3d.environment.sdfgi_cascades = 8

Post-processing steps
---------------------

Some common post-processing steps are listed below.

.. note::

    When using several post-processing steps, try to perform all of them in a
    single FFmpeg command. This will save encoding time and improve quality by
    avoiding multiple lossy encoding steps.

.. _doc_creating_movies_converting_avi:

Converting AVI video to MP4
^^^^^^^^^^^^^^^^^^^^^^^^^^^

While some platforms such as YouTube support uploading the AVI file directly, many
others will require a conversion step beforehand. `HandBrake <https://handbrake.fr/>`__
(GUI) and `FFmpeg <https://ffmpeg.org/>`__ (CLI) are popular open source tools
for this purpose. FFmpeg has a steeper learning curve, but it's more powerful.

The command below converts an AVI video to a MP4 (H.264) video with a Constant
Rate Factor (CRF) of 15. This results in a relatively large file, but is

> well-suited for platforms that will reencode your videos to reduce their size

Reviewer 2: ```suggestion
well-suited for platforms that will re-encode your videos to reduce their size
```

As suggested by CI

  This also makes effects that converge over several frames (such as temporal antialiasing,
  SDFGI and volumetric fog) look better.

.. warning::

    **This feature is not designed for capturing real-time footage during gameplay.**

    Players should use something like `OBS Studio <https://obsproject.com/>`__ or
    `SimpleScreenRecorder <https://www.maartenbaert.be/simplescreenrecorder/>`__
    to record gameplay videos, as they do a much better job at intercepting the
    compositor than Godot can do using Vulkan or OpenGL natively.

    That said, if your game runs at near-real-time speeds when capturing,
    you can still use this feature (but it will lack audible sound playback,
    as sound is saved directly to the video file).

Enabling Movie Maker mode
-------------------------

To enable Movie Maker mode, click the "movie reel" button in the top-right
corner of the editor *before* running the project:

.. image:: img/movie_maker_enable.png

The icon turns red when Movie Maker mode is enabled. Movie Maker status is *not*
persisted when the editor quits, so you must re-enable Movie Maker mode again
after restarting the editor if needed.

.. note::

> Toggling Movie Maker mode while running the project will not have any effect
> until the project is restarted.

Author: ```suggestion
    Toggling Movie Maker mode while running the project will not have any
    effect until the project is restarted.
```
